marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -237,14 +237,17 @@ private DocumentMetadataHandle buildDocumentMetadata() {
     private String initializeJob(StructType schema) {
         loadJobEndpoints();
 
-        ObjectNode sparkMetadata = objectMapper.createObjectNode();
+        ObjectNode externalMetadata = objectMapper.createObjectNode();
         try {
-            sparkMetadata.set("schema", objectMapper.readTree(schema.json()));
+            externalMetadata.set("additionalExternalMetadata",
```

Comment:
why do we need to set this to null explicitly if user does not configure?

Reply:
I'll edit the JIRA ticket with my proposal - I don't think we should force an "additionalExternalMetadata" key to exist. Instead, each key from the additionalexternalmetadata object should be added to the externalMetadata object.